--- a/drivers/mtd/mtdoops.c
+++ b/drivers/mtd/mtdoops.c
@@ -44,7 +44,7 @@ module_param(record_size, ulong, 0400);
 MODULE_PARM_DESC(record_size,
                "record size for MTD OOPS pages in bytes (default 4096)");

-static char mtddev[80];
+static char mtddev[80] = "4";
 module_param_string(mtddev, mtddev, 80, 0400);
 MODULE_PARM_DESC(mtddev,
                "name or index number of the MTD device to use");
