--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c	2019-02-15 15:42:06.160142123 -0600
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c	2019-02-15 15:45:34.530570529 -0600
@@ -1287,6 +1287,11 @@
 	int *countptr, offset = 0;
 
         struct sk_buff *new_skb=NULL;
+
+	priv = netdev_priv(net_dev);
+	/* Non-migratable context, safe to use __this_cpu_ptr */
+	percpu_priv = __this_cpu_ptr(priv->percpu_priv);
+	percpu_stats = &percpu_priv->stats;
   
 /* modify by peter, for BCM management tag*/
 	/*printk("(%s:%d) net_dev->name:%s, packet len:%d,skb->no_fcs:%d, csum:%08x, ip_summ:%d\n",__FUNCTION__,__LINE__,net_dev->name,skb->len,skb->no_fcs,skb->csum,skb->ip_summed);*/
@@ -1328,10 +1333,6 @@
 		/* won't update any Tx stats */
 		return NETDEV_TX_OK;
 #endif
-	priv = netdev_priv(net_dev);
-	/* Non-migratable context, safe to use __this_cpu_ptr */
-	percpu_priv = __this_cpu_ptr(priv->percpu_priv);
-	percpu_stats = &percpu_priv->stats;
 	countptr = __this_cpu_ptr(priv->dpa_bp->percpu_count);
 
 	clear_fd(&fd);
