From 8e51f798df737faae16e58a5d6735eb982e07513 Mon Sep 17 00:00:00 2001
From: StanleyLiu <stanley_liu@accton.com>
Date: Thu, 14 Jun 2018 11:06:44 +0800
Subject: [PATCH 4/8] Change flash size from 256MB to 16MB

---
 board/freescale/t208xrdb/tlb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/freescale/t208xrdb/tlb.c b/board/freescale/t208xrdb/tlb.c
index 2ebea36..af61791 100644
--- a/board/freescale/t208xrdb/tlb.c
+++ b/board/freescale/t208xrdb/tlb.c
@@ -63,7 +63,7 @@ struct fsl_e_tlb_entry tlb_table[] = {
 	/* This will be changed to *I*G* after relocation to RAM. */
 	SET_TLB_ENTRY(1, CONFIG_SYS_FLASH_BASE, CONFIG_SYS_FLASH_BASE_PHYS,
 		      MAS3_SX|MAS3_SR, MAS2_W|MAS2_G,
-		      0, 2, BOOKE_PAGESZ_256M, 1),
+		      0, 2, BOOKE_PAGESZ_16M, 1), /* stanley_liu: change flash size to 16MB */
 
 #ifndef CONFIG_SPL_BUILD
 	/* *I*G* - PCIe 1, 0x80000000 */
-- 
1.7.9.5

