From cfd6487fa67833e43e5334c471ad4572cdcf71a8 Mon Sep 17 00:00:00 2001
From: StanleyLiu <stanley_liu@accton.com>
Date: Thu, 14 Jun 2018 13:45:01 +0800
Subject: [PATCH 08/13] Remove unnecessary message

---
 drivers/ddr/fsl/main.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/ddr/fsl/main.c b/drivers/ddr/fsl/main.c
index d4478eb..a226ac0 100644
--- a/drivers/ddr/fsl/main.c
+++ b/drivers/ddr/fsl/main.c
@@ -100,7 +100,6 @@ static void __get_spd(generic_spd_eeprom_t *spd, u8 i2c_address)
 	if (ret) {
 		printf("PCA: failed to select proper channel\n");
 	}
-	printf("__get_spd: try to read spd from address 0x%x\n", i2c_address);
 
 #ifdef CONFIG_SYS_FSL_DDR4
 	/*
-- 
1.7.9.5

