From a98a15cd44c43ee5af93ec5a4dfa0c46e32bdde4 Mon Sep 17 00:00:00 2001
From: aken_liu <aken_liu@accton.com.tw>
Date: Mon, 6 Aug 2018 10:50:04 +0800
Subject: [PATCH 2/3] fix the order of fan number with the new FAN CPLD

---
 drivers/hwmon/accton_t600_fan.c | 40 ++++++++++++++++++++++++++++++++++------
 1 file changed, 34 insertions(+), 6 deletions(-)

diff --git a/drivers/hwmon/accton_t600_fan.c b/drivers/hwmon/accton_t600_fan.c
index 3e74c37..f116a60 100644
--- a/drivers/hwmon/accton_t600_fan.c
+++ b/drivers/hwmon/accton_t600_fan.c
@@ -469,22 +469,50 @@ static ssize_t fan_show_value(struct device *dev, struct device_attribute *da,
                 ret = sprintf(buf, "%u\n", duty_cycle);
                 break;
             }
-            case FAN_1_INPUT_ATTR_ID(1)...FAN_2_INPUT_ATTR_ID(5):           
-                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[attr->index]));
+            case FAN_1_INPUT_ATTR_ID(1):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_1_INPUT_ATTR_ID(5)]));
                 break;
+            case FAN_1_INPUT_ATTR_ID(2):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_1_INPUT_ATTR_ID(4)]));
+                break;
+            case FAN_1_INPUT_ATTR_ID(3):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_1_INPUT_ATTR_ID(3)]));
+                break;
+            case FAN_1_INPUT_ATTR_ID(4):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_1_INPUT_ATTR_ID(2)]));
+                break;
+            case FAN_1_INPUT_ATTR_ID(5):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_1_INPUT_ATTR_ID(1)]));
+                break;
+            case FAN_2_INPUT_ATTR_ID(1):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_2_INPUT_ATTR_ID(5)]));
+                break;
+            case FAN_2_INPUT_ATTR_ID(2):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_2_INPUT_ATTR_ID(4)]));
+                break;
+            case FAN_2_INPUT_ATTR_ID(3):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_2_INPUT_ATTR_ID(3)]));
+                break;
+            case FAN_2_INPUT_ATTR_ID(4):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_2_INPUT_ATTR_ID(2)]));
+                break;
+            case FAN_2_INPUT_ATTR_ID(5):
+                ret = sprintf(buf, "%u\n", reg_val_to_speed_rpm(data->reg_val[FAN_2_INPUT_ATTR_ID(1)]));
+                break;
+
             case FAN_DIRECTION_ATTR_ID(1)...FAN_DIRECTION_ATTR_ID(5):
                 ret = sprintf(buf, "%d\n", 
                               reg_val_to_direction(data->reg_val[FAN_DIRECTION_REG],
-                              attr->index - FAN_DIRECTION_ATTR_ID(1) ));
+                              FAN_DIRECTION_ATTR_ID(5) - attr->index));
                                                       
                 break;
             case FAN_PRESENT_ATTR_ID(1)...FAN_PRESENT_ATTR_ID(5):
                 ret = sprintf(buf, "%d\n",
                               reg_val_to_is_present(data->reg_val[FAN_PRESENT_REG],
-                              attr->index - FAN_PRESENT_ATTR_ID(1) ));                
+                              FAN_PRESENT_ATTR_ID(5) - attr->index));
                 break;
             case FAN_FAULT_ATTR_ID(1)...FAN_FAULT_ATTR_ID(5):
-                ret = sprintf(buf, "%d\n", is_fan_failed(data, attr->index - FAN_FAULT_ATTR_ID(1) ));
+                ret = sprintf(buf, "%d\n", is_fan_failed(data, FAN_FAULT_ATTR_ID(5) - attr->index));
                 break;
             case CPLD_VERSION:
                 ret = sprintf(buf, "%d\n", data->reg_val[CPLD_VERSION]);
@@ -493,7 +521,7 @@ static ssize_t fan_show_value(struct device *dev, struct device_attribute *da,
             case FAN_ENABLE_ATTR_ID(1)...FAN_ENABLE_ATTR_ID(5):
                 ret = sprintf(buf, "%d\n",
                               reg_val_to_is_enable(data->reg_val[FAN_ENABLE_REG],
-                              attr->index - FAN_ENABLE_ATTR_ID(1) ));
+                              FAN_ENABLE_ATTR_ID(5) - attr->index));
                 break;
 
             case FAN_WATCHDOG:
-- 
1.9.1

