diff --git a/drivers/net/phy/mdio_bus.c b/drivers/net/phy/mdio_bus.c
index 9c86de2..0397fb5 100644
--- a/drivers/net/phy/mdio_bus.c
+++ b/drivers/net/phy/mdio_bus.c
@@ -371,6 +371,34 @@ struct phy_device *mdiobus_scan(struct mii_bus *bus, int addr)
 }
 EXPORT_SYMBOL(mdiobus_scan);
 
+static int mdiobus_match_name( struct device * dev, void * data )
+{
+    const char * name = data;
+
+    return sysfs_streq( name, dev_name( dev ) );
+}
+
+/**
+ * mdiobus_find_by_name - Convenience function for retrieving an mii_bus pointer
+ *                        by name
+ * @name: name of the bus being searched for
+ */
+struct mii_bus * mdiobus_find_by_name( char * name )
+{
+    struct device * dev;
+
+    /* search devices registered for with the mdio_bus_class using the device
+       name as the matching criteria */
+    dev = class_find_device( &mdio_bus_class,
+                             NULL,
+                             (void *)name,
+                             mdiobus_match_name );
+
+    /* return the mii_bus pointer or NULL if none was found */
+    return dev ? container_of( dev, struct mii_bus, dev ) : NULL;
+}
+EXPORT_SYMBOL( mdiobus_find_by_name );
+
 /**
  * mdiobus_read - Convenience function for reading a given MII mgmt register
  * @bus: the mii_bus struct
-- 
1.8.3.1

