diff -Naur a/arch/powerpc/sysdev/fsl_soc.c b/arch/powerpc/sysdev/fsl_soc.c
--- a/arch/powerpc/sysdev/fsl_soc.c	2020-01-14 13:43:17.985946212 -0600
+++ b/arch/powerpc/sysdev/fsl_soc.c	2020-01-14 13:48:48.368788752 -0600
@@ -195,7 +195,7 @@
 /*!
 ** @brief Invoked at startup. Will allocate and map registers used during reset.
 **        linux reset     -> FNC Warm reset
-**        linux halt      -> FNC Hard Reset
+**        linux halt      -> FNC Power off
 **        linux power off -> FNC Hard Reset
 */
 extern void fss2_reset_warm(char *cmd);
@@ -204,9 +204,9 @@
 
 static int __init setup_finity_reset(void)
 {
-    ppc_md.restart    = fss2_reset_warm;         // Warm (default)
-    pm_power_off      = fss2_reset_power_off;    // Hard
-    ppc_md.halt       = fss2_reset_hard;         // Hard
+    ppc_md.restart    = fss2_reset_warm;         // Warm (reboot) (default)
+    pm_power_off      = fss2_reset_hard;         // Hard reset
+    ppc_md.halt       = fss2_reset_power_off;    // Power off
 
 	return 0;
 }
