From b6a9bfa6d62fdf3c1cfde93052611f2bffe376be Mon Sep 17 00:00:00 2001
From: roy_chuang <roy_chuang@edge.core.com>
Date: Tue, 24 Sep 2019 19:50:29 +0800
Subject: [PATCH] Modify default time-interval from 3 to 6 seconds.

---
 drivers/hwmon/accton_t600_fan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/accton_t600_fan.c b/drivers/hwmon/accton_t600_fan.c
index 1bf9619..b7c2480 100755
--- a/drivers/hwmon/accton_t600_fan.c
+++ b/drivers/hwmon/accton_t600_fan.c
@@ -51,7 +51,7 @@ enum fan_duty_cycle {
 #include <linux/kthread.h>
 
 #define FAN_SPEED_CTRL_INTERVAL	3000
-#define FAN_SPEED_DELAY_DEFAULT	3
+#define FAN_SPEED_DELAY_DEFAULT	6
 #define DUTY_CYCLE_NONE         -1
 #define THERMAL_SENSOR_DATA_LEN 6
 
-- 
1.9.1

